code/datums/components/durability.dm

```diff
@@ -0,0 +1,189 @@
+	broken_w_class
+)
+	if (!isnull(broken))
+		src.broken = broken
+
+	if (!isnull(broken_prefix))
+		src.broken_prefix = broken_prefix
+
+	if (!isnull(broken_name))
+		src.broken_name = broken_name
+
+	if (!isnull(broken_desc))
+		src.broken_desc = broken_desc
+
+	if (!isnull(max_durability))
+		src.max_durability = max_durability
+		src.current_durability = max_durability
+
+	if (!isnull(current_durability))
+		src.current_durability = current_durability
+
+	if (!isnull(broken_icon_state))
+		src.broken_icon_state = broken_icon_state
+
+	if (!isnull(break_sound))
+		src.break_sound = break_sound
+
+	if (!isnull(broken_force))
+		src.broken_force = broken_force
+
+	if (!isnull(broken_throw_force))
+		src.broken_throw_force = broken_throw_force
+
+	if (!isnull(broken_throw_force_decrease))
+		src.broken_throw_force_decrease = broken_throw_force_decrease
+
+	if (!isnull(broken_throw_range))
+		src.broken_throw_range = broken_throw_range
+
+	if (!isnull(broken_embedding))
+		src.broken_embedding = broken_embedding
+
+	if (!isnull(broken_block_chance))
+		src.broken_block_chance = broken_block_chance
+
+	if (!isnull(broken_message))
+		src.broken_message = broken_message
+
+	if (!isnull(broken_w_class))
+		src.broken_w_class = broken_w_class
+
+
+/datum/component/durability/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_ITEM_AFTERATTACK, PROC_REF(on_after_attack))
+
+/datum/component/durability/UnregisterFromParent()
+	UnregisterSignal(parent, list(COMSIG_ITEM_AFTERATTACK))
+
+/datum/component/durability/proc/on_after_attack(obj/item/source, atom/target, mob/living/user, proximity)
+	SIGNAL_HANDLER
+
+	if(broken)
+		return NONE
+
+	current_durability -= 1
+
+	if(!isnull(max_durability) && current_durability == 0)
+		broken = TRUE
+
+		if (!isnull(broken_prefix))
+			source.name = "[broken_prefix] [source.name]"
+
+		if (!isnull(broken_name))
+			source.name = broken_name
+
+		if (!isnull(broken_desc))
+			source.desc = broken_desc
+
+		if (!isnull(broken_icon_state))
+			source.icon_state = broken_icon_state
+			source.inhand_icon_state = broken_icon_state
+			source.worn_icon_state = broken_icon_state
+
+		if (!isnull(broken_force))
+			source.force = broken_force
+
+		if (!isnull(broken_throw_force))
+			source.throwforce = broken_throw_force
+
+		if (!isnull(broken_throw_range))
+			source.throw_range = broken_throw_range
+
+		if (!isnull(broken_embedding))
+			source.embedding = broken_embedding
+
+		if (!isnull(broken_block_chance))
+			source.block_chance = broken_block_chance
+
+		if (!isnull(broken_w_class))
+			source.w_class = broken_w_class
+
+		if (!isnull(broken_force_decrease))
+			source.force -= broken_force_decrease
+
+		if (!isnull(broken_throw_force_decrease))
+			source.throwforce -= broken_throw_force_decrease
+
+		if (!isnull(broken_message))
+			source.visible_message(broken_message)
+
+		if (!isnull(break_sound))
+			playsound(source, break_sound, 100, TRUE)
+
+	return COMPONENT_AFTERATTACK_PROCESSED_ITEM
```

Comment:
You should also delete the component here (`qdel(src)`) instead of tracking `broken`, we don't need it any more.